### PR TITLE
Fix https://jira.sonarsource.com/browse/TBSTFNP-46 - on PR builds, ru…

### PR DESCRIPTION
**Pull request by bgavrilMS**
…n the SQ analysis in issues / incremental mode

### CODE REVIEW

File: Tasks/Common/SonarQubeHelpers/SonarQubeHelper.ps1 [hunk 1]
```diff
@@ -202,16 +201,21 @@ function Assert
     }
 }
 
-
 #
-# Returns true if this build was triggered in response to a PR 
+# Returns true if this build was triggered in response to a PR
+#
+# Remark: this logic is temporary until the platform provides a more robust way of deteriming PR builds
```
Comment (bgavrilMS) [inline Tasks/Common/SonarQubeHelpers/SonarQubeHelper.ps1]: ETA is S97

File: Tasks/SonarQubePostTest/CodeAnalysisFilePathComputation.ps1 [hunk 2]
```diff
@@ -1,231 +0,0 @@
-. ./SonarQubePostTestImpl.ps1
```
Comment (bgavrilMS) [inline Tasks/SonarQubePostTest/CodeAnalysisFilePathComputation.ps1]: At this point this is dead code as it's handling the analysis report for PR-CA. I'll bring it back as part of the PR-CA work.

File: Tasks/SonarQubePreBuild/SonarQubePreBuildImpl.ps1 [hunk 3]
```diff
@@ -138,9 +138,6 @@ function UpdateArgsForPullRequestAnalysis($cmdLineArgs)
         {
             $cmdLineArgs = $cmdLineArgs + " " + "/d:sonar.analysis.mode=incremental"
         }
-
-		#use this variable in post-test task
-		SetTaskContextVariable "MSBuild.SonarQube.AnalysisModeIsIncremental" "true"
```
Comment (bgavrilMS) [inline Tasks/SonarQubePreBuild/SonarQubePreBuildImpl.ps1]: Useless code, but in any case we can just check if the build is a PR build in the post task and not use this mechanism

File: Tasks/Common/SonarQubeHelpers/SonarQubeHelper.ps1 [hunk 4]
```diff
@@ -202,16 +201,21 @@ function Assert
...
 function IsPrBuild
 {
-    $prcaEnabledString = GetTaskContextVariable "PullRequestSonarQubeCodeAnalysisEnabled"
-    $isPrBuild = Convert-String $prcaEnabledString Boolean
+    $sourceBranch = $env:Build_SourceBranch
 
-    return $isPrBuild;
+    if ([string]::IsNullOrWhiteSpace($sourceBranch))
+    {
+        throw "Could not determine the source branch, please log a bug"
```
Comment (duncanpMS) [inline Tasks/Common/SonarQubeHelpers/SonarQubeHelper.ps1]: Where?

File: Tasks/Common/SonarQubeHelpers/SonarQubeHelper.ps1 [hunk 5]
```diff
@@ -202,16 +201,21 @@ function Assert
...
+    }
+    
+    return $sourceBranch.StartsWith("refs/pull/", [StringComparison]::OrdinalIgnoreCase)
```
Comment (duncanpMS) [inline Tasks/Common/SonarQubeHelpers/SonarQubeHelper.ps1]: Is this the same for git and TfFV?

**Comment by duncanpMS** [comment]
LGTM

File: Tasks/Common/SonarQubeHelpers/SonarQubeHelper.ps1 (hunk 4 above)
Comment (bgavrilMS) [inline Tasks/Common/SonarQubeHelpers/SonarQubeHelper.ps1]: I removed that part of the message

File: Tasks/Common/SonarQubeHelpers/SonarQubeHelper.ps1 (hunk 5 above)
Comment (bgavrilMS) [inline Tasks/Common/SonarQubeHelpers/SonarQubeHelper.ps1]: A very good point. PRs are only relevant on TfsGit at this point. I am adding an extra check for the Repository Provider.

**Pull request merged**